src/test/java/uk/nhs/tis/sync/service/DataRequestServiceTest.java

```diff
@@ -990,4 +990,48 @@ void shouldReturnEmptyWhenLocalOfficeMessageHasWrongKey() {
     assertThat("Unexpected DTO count.", dbcs.size(), is(0));
     verifyNoInteractions(referenceService);
   }
+
+  @Test
+  void shouldReturnGmcDetailsWhenGmcDetailsFound() {
+    GmcDetailsDTO expectedDto = new GmcDetailsDTO();
+    when(tcsService.findGmcDetailsIn(Collections.singletonList("40")))
+        .thenReturn(Collections.singletonList(expectedDto));
+
+    Map<String, String> message = new HashMap<String, String>() {{
+      put("table", "GmcDetails");
+      put("id", "40");
+    }};
+    List<Object> retrievedDtos = service.retrieveDtos(message);
+
+    assertThat("Unexpected DTO count.", retrievedDtos.size(), is(1));
+    assertThat("Unexpected DTO.", retrievedDtos.get(0), sameInstance(expectedDto));
+  }
+
+  @Test
+  void shouldReturnEmptyWhenGmcDetailsNotFound() {
+    when(tcsService.findGmcDetailsIn(Collections.singletonList("40")))
+        .thenReturn(null);
```

Comment:
I'll assume `null` is correct here, rather than an empty list.

Reply:
I think so. The DataRequestService uses `createNonNullList()` to handle these nulls and turn them into an empty list in `retrieveDtosById()`